### PR TITLE
[Triton/Gluon] Fix int32 overflow in _gluon_deepgemm_fp8_paged_mqa_logits_preshuffle_varctx K/scale load offset

### PR DESCRIPTION
**Pull request by wufann**
## Motivation
The VarCtx gluon kernel `_gluon_deepgemm_fp8_paged_mqa_logits_preshuffle_varctx`
loads K/scale with `gl.amd.cdna3.buffer_load`, whose voffset is a **32-bit**
element offset. The block index (`context_kv_idx`, int32) is multiplied by
`stride_k_seq` while still int32, so for a wide paged cache the product
`block_index * stride_k_seq` crosses `2**31` and wraps. The load then reads out of
bounds and returns 0, so whole logit segments silently collapse to `0` for the
sequences that own high block indices (large batch). This PR loads K/scale with
64-bit pointer arithmetic via `gl.load`, matching the non-VarCtx preshuffle path.

### Issue
`deepgemm_fp8_paged_mqa_logits(..., Preshuffle=True, VarCtxSchedule=<schedule>)`
produces wrong output — whole segments become exactly `0.0` — at large batch with
variable context lengths on MI355X / gfx950. The non-VarCtx SplitKV path
(`VarCtxSchedule=None`) is always correct.

### Root cause
`stride_k_seq = kv_cache_fp8.stride(0) = KVBlockSize * index_dim`. The paged cache
row is `KVBlockSize * (head_dim + 4)` bytes — the extra 4 fp32-scale bytes per
block make the row stride **non-power-of-two**, e.g. `64 * (128 + 4) = 8448`
(not `64*128 = 8192`).

`context_kv_idx` is int32, and `gl.amd.cdna3.buffer_load` takes a **32-bit**
offset. The offset `context_kv_idx * stride_k_seq` therefore overflows once

```
block_index * 8448 >= 2**31  ->  block_index >= 2**31 / 8448 = 254201
```

The wrapped offset lands outside the buffer, the hardware buffer bound makes the
load return 0, and the resulting logits are `relu(q·0)·w = 0`.

## Technical Details

For each of the 4 K/scale load pairs in the `LoadBlockIndiceForEachStage == True`
branch (the `KVBlockSize=64` path used by the c4 indexer), switch from
`buffer_load(ptr=..., offsets=... int32 ...)` to `gl.load(ptr + ... int64 ...)`:

```python
# before (int32 offset -> overflow at block_index*stride_k_seq >= 2**31):
k_next_0 = gl.amd.cdna3.buffer_load(
    ptr=KV_buffer,
    offsets=offset_k_fixed + context_kv_idx_next_0[None, :] * stride_k_seq,
)
# after (64-bit pointer arithmetic, cannot overflow):
k_next_0 = gl.load(
    KV_buffer
    + offset_k_fixed
    + context_kv_idx_next_0.to(gl.int64)[None, :] * stride_k_seq
)
```

Same change for `k_next_1` and the two `k_scale_f_next_*`
(`context_kv_idx.to(gl.int64) * stride_scale_seq`). This mirrors exactly what the
non-VarCtx `_gluon_deepgemm_fp8_paged_mqa_logits_preshuffle` kernel already does.
## Test Plan
`op_tests/test_pa_mqa_logits_varctx_overflow.py` builds a distinct-per-sequence
paged cache (`block_tables = arange`) with `batch=256` and variable ctx so the
global block index crosses 254201, asserts the config actually reaches the
`2**31` boundary, then checks the VarCtx output against the known-good SplitKV
path: (1) no valid position is zeroed where SplitKV is non-zero, and (2) per-row
`cos(VarCtx, SplitKV) > 0.999`.

## Test Result
Test on (gfx950 / MI355X)
| kernel | test |
|---|---|
| before (buffer_load, int32) | **FAIL** — `VarCtx zeroed 51576 valid logits in row 254` |
| after (gl.load, int64) | **PASS** |

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: op_tests/test_pa_mqa_logits_varctx_overflow.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (Boss2002n) [reply op_tests/test_pa_mqa_logits_varctx_overflow.py]: @wufann

[the remark this answers is not in the document]
Comment (wufann) [reply op_tests/test_pa_mqa_logits_varctx_overflow.py]: > @wufann

Hi, move ut to op_tests/triton_tests/attention

File: op_tests/test_pa_mqa_logits_varctx_overflow.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+This test uses distinct per-sequence blocks and a batch big enough to cross
+2**31, then asserts the VarCtx output matches the known-good SplitKV output and
+that no valid position was zeroed.
+"""
+
+import pytest
+import torch
+
+from aiter.ops.shuffle import shuffle_weight
```
[the remark this answers is not in the document]
Comment (wufann) [reply op_tests/test_pa_mqa_logits_varctx_overflow.py]: fixed

File: op_tests/test_pa_mqa_logits_varctx_overflow.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        Preshuffle=True,
+        KVBlockSize=KV_BLOCK,
+        WavePerEU=2,
+        VarCtxSchedule=vcs,
+    )
+    torch.cuda.synchronize()
+    return out
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="requires a ROCm GPU")
```
[the remark this answers is not in the document]
Comment (wufann) [reply op_tests/test_pa_mqa_logits_varctx_overflow.py]: fixed
